include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,346 @@
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    soa_vector() : storage(0) {}
+
+    //! Constructs the container with `n` default-inserted instances of `Type`. No copies are made.
+    soa_vector ( std::size_t n) : storage(n) {}
+
+    //! Constructs the container with `n` copies of elements with `value` value.
+    soa_vector(std::size_t n, value_type value)  : storage(n, value) {}
+
+    //! Constructs the container with the contents of the initializer list `l`.
+    soa_vector( std::initializer_list<Type> l ) : storage(l.size())
+    {
+      auto ptr = l.begin();
+      for(std::size_t i = 0;i < size(); ++i)
+      {
+        kumi::for_each( [&](auto m, auto& v) { v[i] = m; }, kumi::flatten_all(*ptr++), storage );
+      }
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Capacity
+    //! @{
+    //==============================================================================================
+    //! Returns the number of elements in the container
+    EVE_FORCEINLINE std::size_t size()  const noexcept { return storage.size_; }
+
+    //! Returns the number of elements that the container has currently allocated space for
+    EVE_FORCEINLINE std::size_t capacity()  const noexcept { return storage.capacity_; }
+
+    //! Requests the removal of unused capacity.
+    // EVE_FORCEINLINE void shrink_to_fit() noexcept
+    // {
+    //   kumi::for_each([](auto& m) { m.shrink_to_fit(); }, storage);
+    // }
+
+    //! Checks if the container has no elements
+    EVE_FORCEINLINE bool empty() const noexcept { return size() == 0ULL; }
+
+    //! @brief Increase the capacity of the vector to a value that's greater or equal to `n`.
+    //! If `n` is greater than the current capacity(), new storage is allocated and all iterators,
+    //! including the past-the-end iterator, and all references to the elements are invalidated.
+    //! Otherwise, the method does nothing.
+    //! @warning reserve() does not change the size of the vector.
+    //! @param n New capacity of the `eve::soa_vector`
+    void reserve(std::size_t n)
+    {
+      if(n > capacity())
+      {
+        auto const sz = size();
+        soa_vector that(n);
+        storage_type::piecewise_copy(that.storage, storage, sz);
```

Comment:
Ah - I see why you needed `piecewise_copy`.
Please use `eve::algo::copy`

Reply:
Noted.

Reply:
Fixed.